### ISSUE TITLE
Release v0.30.0 no longer emits traces_receiver_* metrics

### ISSUE THREAD

**Issue by modulitos** [grafana/agent#2644]
### Repro:
Start the `grafana/agent:v0.30.0` image with `/bin/agent -config.file=/etc/agent/agent.yaml -server.http.address=0.0.0.0:80 -config.expand-env`.

Here's my `/etc/agent/agent.yaml`, but I'm not sure if the specifics of this file matters for this bug:
```
server:
    log_level: info
    log_format: json
traces:
    configs
      - name: zipkin_traces
        batch:
            send_batch_size: 1000
            timeout: 5s
        receivers:
            zipkin:
                endpoint: 0.0.0.0:9411
        remove_write:
            //etc
        tail_sampling:
            //etc
```
Then run `curl localhost:80/metrics | grep receiver` to see the receiver metrics.

### Expected
We should see the `traces_receiver_accepted_spans` and `traces_receiver_accepted_spans` metrics in the output. For example:

```
# HELP traces_receiver_accepted_spans Number of spans successfully pushed into the pipeline.
# TYPE traces_receiver_accepted_spans counter
traces_receiver_accepted_spans{receiver="zipkin",traces_config="zipkin_traces",transport="http_v2_json"} 827851
# HELP traces_receiver_refused_spans Number of spans that could not be pushed into the pipeline.
# TYPE traces_receiver_refused_spans counter
traces_receiver_refused_spans{receiver="zipkin",traces_config="zipkin_traces",transport="http_v2_json"} 0
```
This is the case when I do all of the above, but with image `grafana/agent:v0.29.0` instead.

### Actual
There is no output for any metrics matching `receiver` on v0.30.0.

---

It seems like these `traces_receiver_*` metrics are expected - at least according to the Troubleshooting docs here: https://grafana.com/docs/tempo/latest/troubleshooting/agent/

Could this be a bug from the `v0.30.0` release?

**Comment by tpaschalis** [comment]
Hey there, thanks for the report! 👋 

We have identified the issue about being about the way that our OpenTelemetry Collector dependency exposes these metrics in their v0.63. We'll share news about a fix in due time.

**Comment by tpaschalis** [comment]
Closing this, as it has been fixed with the v0.30.1 patch release.

Again, thanks for bringing this to our attention!

**Comment by modulitos** [comment]
@tpaschalis I can confirm this bug is fixed. Thank you for the quick fix!
